src/browser-pool.ts

```diff
@@ -632,7 +639,9 @@ export class BrowserPool<
             // If the hooks or the launch fails, we need to delete the controller,
             // because otherwise it would be stuck in limbo without a browser.
             await this._executeHooks(this.preLaunchHooks, pageId, launchContext);
+            tryCancel();
             const browser = await browserPlugin.launch(launchContext);
+            tryCancel();
```

Comment:
The thrown error will be rethrown in the `catch`. I'm just checking that this is the desired behavior. Seems good to me.

Reply:
Yop, that is what we want, it needs to bubble up to the `addTimeoutToPromise()` where it will be ignored.